WhisperAPI/WhisperAPI/Services/Suggestions/SuggestionsService.cs

```diff
@@ -144,6 +145,35 @@ public bool UpdateContextWithSelectedSuggestion(ConversationContext conversation
             return false;
         }
 
+        internal async Task<IEnumerable<Recommendation<Document>>> GetQuerySearchRecommendations(ConversationContext conversationContext)
+        {
+            var allParsedRelevantQueries = conversationContext.ContextItems
+                .Where(c => c.SearchQuery.Type == SearchQuery.MessageType.Customer && c.Relevant)
+                .Select(x => x.NlpAnalysis.ParsedQuery).ToList();
+
+            var allWords = string.Join(" ", allParsedRelevantQueries.ToArray()).Split(" ").AsEnumerable();
```

Comment:
This comment goes for this line and the one below that. You should only iterate on parsed queries you need instead of taking all parsed queries and splitting into words.

But most importantly, you should avoid duplicates. Create a HashSet<string> (that uses a string.ignorecase comparer) and add words in it while the hashset count is less than 7. You can iterate on all custommer parsedqueries.reverse(), in that loop split the words, add them in the hashset and if you have the ddsired count, break.